### DISCUSSION TITLE
.Net Core 3.0 cannot load custom addin

### DISCUSSION THREAD

**Discussion by markofranjic** [cake-build/cake#3090] [Q&A] [answered]
Hi Team,

Hi Team,

I have update my Cake Version to 0.35 because 0.35 has support for .net core 3.0.
I'v tried write my own addin for Cake which will support System.Text.Json;
Example:
[CakeMethodAlias]	
public static T DeserializeJson<T>(this ICakeContext context, string json)
{
	var options = new JsonSerializerOptions		
	{	
		AllowTrailingCommas = true
	};
	return JsonSerializer.Deserialize<T>(json, options);
}

When I referenced my .dll to Cake script and try to run Task I got
Could not load C:\Users\<>\source\repos\Cake.Json\Cake.Json\bin\Debug\netcoreapp3.0\Cake.Json.dll (missing System.Runtime, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
Also, I have tried .net standard 2.1 but doesn't work

Could you please help me with that?

**Comment by devlead** [comment, marked as answer]
System.Text.Json assembly isn't currently automatically loaded under .NET Core 3, but you can load it, example:

```cake
 #r "System.Text.Json"

public class WeatherForecast
{
    public DateTimeOffset Date { get; set; }
    public int TemperatureC { get; set; }
    public string Summary { get; set; }
}

static string SerializeAsJson<T>(this T value)
{
    return System.Text.Json.JsonSerializer.Serialize<T>(value);
}

var weatherForecast = new WeatherForecast{
    Date = DateTimeOffset.UtcNow,
    TemperatureC = 11,
    Summary ="Cloudy"
};

var json = weatherForecast.SerializeAsJson();

Information(json);
```
will output
```json
{"Date":"2019-11-04T13:03:47.0779982+00:00","TemperatureC":11,"Summary":"Cloudy"}
```

To use it on .NET 4.8 or .NET Core 2.1 you'll need to fetch dependency from nuget using addin syntax
```cake
#addin nuget:?package=System.Text.Json&version=4.6.0&loaddependencies=true
```
